packages/tools/devtools/devtools-core/src/messaging/container-devtools-messages/DataVisualization.ts

```diff
@@ -20,6 +20,25 @@ export namespace DataVisualization {
 	 */
 	export const MessageType = "DATA_VISUALIZATION";
 
+	/**
+	 * Reasons why a DataVisualization message is being sent.
+	 *
+	 * @internal
+	 */
+	export const enum UpdateReason {
+		/**
+		 * Visualization was requested by user interaction (e.g., clicking to expand tree).
+		 * Should not trigger UI animations.
+		 */
+		UserRequested = "userRequested",
+
+		/**
+		 * Visualization updated due to actual data changes in the underlying shared object.
+		 * Should trigger UI animations to indicate data has changed.
+		 */
```

Comment:
Nit: while our `devtools-view` is currently the only library that listens for these events, we've generally architected `devtools-core` to try to be agnostic to how it is consumed. For that reason, I would prefer to avoid the notes here explaining how these flags are used. I think it is enough to just describe what they mean.
```suggestion
		/**
		 * Visualization was requested by user interaction (e.g., clicking to expand tree).
		 */
		UserRequested = "userRequested",

		/**
		 * Visualization updated due to actual data changes in the underlying shared object.
		 */
```